include/Graph/Graph.h

```diff
@@ -67,6 +67,10 @@ void Graph<Node>::insert(Node *Src, Node *Dest, int Left, int Right) {
         this->insert(Src, DestDest);
       }
     }
+  } else if(Left == 2 && Right == 0){
```

Comment:
Do you have an example for a situation where the left will be 2 and the right will be 0? 
I think we do not need to change anything in the graph because we are already considering nullptr as a heap location.

Reply:
For 
```C
*p=NULL;
```
llvm is generating
```
store i32* null, i32** %9, align 8, !dbg !32
```
in which relative level of redirecting is (2,1) but because NULL is heap location we are changing its redirection to 0 then we are getting (2,0) and we have to handle it separately.

Reply:
Interesting, thanks!